auth/src/main/java/org/aerogear/mobile/security/checks/ScreenLockCheck.java

```diff
@@ -0,0 +1,31 @@
+package org.aerogear.mobile.security.checks;
+
+import android.app.KeyguardManager;
+import android.content.Context;
+import android.os.Build;
+
+import org.aerogear.mobile.security.SecurityCheck;
+import org.aerogear.mobile.security.SecurityCheckResult;
+import org.aerogear.mobile.security.impl.SecurityCheckResultImpl;
+
+/**
+ * A check for whether the device the application is running on has a screen lock.
+ */
+public class ScreenLockCheck implements SecurityCheck {
+    private static final String NAME = "detectScreenLock";
+
+    /**
+     * Check whether the device has an automatic screenlock
+     * @param context Context to be used by the check.
+     * @return <code>true</code> if the device has an automatic screen lock
+     */
+    @Override
+    public SecurityCheckResult test(Context context){
+        if(Build.VERSION.SDK_INT >= Build.VERSION_CODES.M){
```

Comment:
what is this check for ? and why is it needed ? perhaps a comment here helps

Reply:
@matzew @ciaranRoche The Keyguard Manager is only available from Android M onwards. I think we should make a note in the code here about this. We will also need something in the docs too to mention this.

Reply:
Ah, my bad. Trying to remember from memory,